src/datepicker/utils/date-helpers.js

```diff
@@ -25,6 +48,263 @@ class DateHelpers<T> {
+    return this.getAdapterWithNewLocale(locale)
+      .formatByString(date, 'EEEE')
+      .charAt(0);
+  };
+  getMonthInLocale: (number, locale: mixed) => string = (
+    monthNumber,
+    locale,
+  ) => {
+    const localeAdapter = this.getAdapterWithNewLocale(locale);
+    return localeAdapter.formatByString(
+      localeAdapter.setMonth(localeAdapter.date(), monthNumber),
+      'LLLL',
+    );
+  };
+  getWeekdayInLocale: (T, locale: mixed) => string = (date, locale) => {
+    return this.getAdapterWithNewLocale(locale).formatByString(date, 'EEEE');
+  };
+  getEndOfWeek: T => T = date => {
+    return this.adapter.endOfWeek(date);
+  };
+  getDay: T => number = date => {
+    return Number(this.adapter.formatByString(date, 'e')) - 1;
+  };
+  addWeeks: (T, number) => T = (date, weekNumber) => {
+    return this.adapter.addDays(date, weekNumber * 7);
+  };
+  subWeeks: (T, number) => T = (date, weekNumber) => {
+    return this.addWeeks(date, weekNumber * -1);
+  };
+  addYears: (T, number) => T = (date, yearNumber) => {
+    return this.adapter.addMonths(date, yearNumber * 12);
+  };
+  subYears: (T, number) => T = (date, yearNumber) => {
+    return this.addYears(date, yearNumber * -1);
+  };
+  isSameYear: (?T, ?T) => boolean = (fromDate, toDate) => {
+    if (fromDate && toDate) {
+      return this.adapter.isSameYear(fromDate, toDate);
+    }
+    return false;
+  };
+  isStartOfMonth: T => boolean = date => {
+    return this.adapter.isSameDay(date, this.adapter.startOfMonth(date));
+  };
+  isEndOfMonth: T => boolean = date => {
+    return this.adapter.isSameDay(date, this.adapter.endOfMonth(date));
+  };
+  isDayInRange: (T, T, T) => boolean = (date, startDate, endDate) => {
+    return this.adapter.isWithinRange(date, [startDate, endDate]);
+  };
+  isSameDay: (?T, ?T) => boolean = (fromDate, toDate) => {
+    if (fromDate && toDate) {
+      return this.adapter.isSameDay(fromDate, toDate);
+    }
+    return false;
+  };
+  isSameMonth: (?T, ?T) => boolean = (fromDate, toDate) => {
+    if (fromDate && toDate) {
+      return this.adapter.isSameMonth(fromDate, toDate);
+    }
+    return false;
+  };
+  differenceInCalendarDays: (T, T) => number = (fromDate, toDate) => {
+    const msDiff = this.adapter.getDiff(fromDate, toDate);
+    return msDiff / 864e5;
+  };
+  subDays: (T, number) => T = (date, days) => {
+    return this.adapter.addDays(date, days * -1);
+  };
+  subMonths: (T, number) => T = (date, months) => {
+    return this.adapter.addMonths(date, months * -1);
+  };
+  min: (Array<T>) => T = dates => {
+    return dates.reduce((minDate, date) => {
+      return this.adapter.isBefore(date, minDate) ? date : minDate;
+    });
+  };
+  max: (Array<T>) => T = dates => {
+    return dates.reduce((maxDate, date) => {
+      return this.adapter.isAfter(date, maxDate) ? date : maxDate;
+    });
+  };
+  getEffectiveMinDate: ({minDate?: T, includeDates?: Array<T>}) => T = ({
```

Comment:
`minDate` and `includeDates` used to not be optional, but then they include all of this code for checking if they're defined.  I decide to make them optional just so I could cover all of that code and not have it trigger flow errors, the other option would be to actually take that code out if it's really not optional.